tests/system/test_cluster/test_agent_groups/test_group_sync_status.py

```diff
@@ -0,0 +1,171 @@
+from wazuh_testing.tools.system import HostManager
+from system import (assign_agent_to_new_group, create_new_agent_group, delete_agent_group, execute_wdb_query,
+                    restart_cluster)
+from wazuh_testing.tools.configuration import get_test_cases_data
+from system.test_cluster.test_agent_groups.common import register_agent
+pytestmark = [pytest.mark.cluster, pytest.mark.enrollment_cluster_env]
+
+test_infra_hosts = ['wazuh-master', 'wazuh-worker1', 'wazuh-worker2', 'wazuh-agent1', 'wazuh-agent2']
+test_infra_managers = ['wazuh-master', 'wazuh-worker1', 'wazuh-worker2']
+test_infra_agents = ['wazuh-agent1', 'wazuh-agent2']
+groups = ['group_master', 'group_worker1', 'group_worker2']
+workers = ['wazuh-worker1', 'wazuh-worker2']
+groups_created = []
+query = "global 'sql select name, group_sync_status from agent;'"
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                              'provisioning', 'enrollment_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+data_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
+local_path = os.path.dirname(os.path.abspath(__file__))
+test_cases_yaml = os.path.join(data_path, 'cases_group_sync.yaml')
+wdb_query = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'script/wdb-query.py')
+agent_conf_file = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                               '..', '..', 'provisioning', 'enrollment_cluster', 'roles', 'agent-role', 
+                               'files', 'ossec.conf')
+t1_configuration_parameters, t1_configuration_metadata, t1_case_ids = get_test_cases_data(test_cases_yaml)
+
+@pytest.fixture()
+def group_creation_and_assignation(metadata, target_node):
+
+    agent_ids = []
+    for agent in test_infra_agents:
+        agent_ip, agent_id, agent_name, manager_ip = register_agent(agent, test_infra_hosts[0], host_manager)
+        time.sleep(T_025)
+        agent_ids.append(agent_id)
+
+    restart_cluster(test_infra_agents, host_manager)
+
+    time.sleep(T_1)
+    for group in groups:
+        create_new_agent_group(target_node, group, host_manager)
+
+    if metadata['agent_in_group'] == 'agent1':
+        for group in groups:
+            assign_agent_to_new_group(target_node, group, agent_ids[0], host_manager)
+
+    elif metadata['agent_in_group'] == 'agent2':
+        for group in groups:
+            assign_agent_to_new_group(target_node, group, agent_ids[1], host_manager)
+
+    else:
+        for group in groups:
+            for agent_id in agent_ids:
+                assign_agent_to_new_group(target_node, group, agent_id, host_manager)
+
+    yield
+
+    for group in groups:
+        delete_agent_group(test_infra_hosts[0], group, host_manager, 'api')
+
+
+@pytest.fixture()
+def wait_end_initial_syncreq():
+    result = execute_wdb_query(query, test_infra_hosts[0], host_manager)
+    while 'syncreq' in result:
```

Comment:
We need to add a timeout

Reply:
Timeout created in https://github.com/wazuh/wazuh-qa/pull/4143/commits/6c2985d1a2e92877c3130604cdbe41d533b20a75